homeassistant/components/zha/api.py

```diff
@@ -74,6 +76,38 @@
 }
 
 
+@websocket_api.require_admin
+@websocket_api.async_response
+@websocket_api.websocket_command({
+    vol.Required('type'): 'zha/devices/add'
+})
+async def websocket_add_devices(hass, connection, msg):
+    """Add ZHA zigbee devices."""
+    zha_gateway = hass.data[DATA_ZHA][DATA_ZHA_GATEWAY]
+
+    async def forward_messages(data):
+        """Forward events to websocket."""
+        connection.send_message(websocket_api.event_message(msg['id'], data))
+
+    remove_dispatcher_function = async_dispatcher_connect(
+        hass,
+        "zha_gateway_message",
+        forward_messages
+    )
+
+    @callback
+    def async_cleanup() -> None:
+        """Remove signal listener and turn off debug mode."""
+        zha_gateway.async_disable_debug_mode()
+        remove_dispatcher_function()
+
+    connection.subscriptions[msg['id']] = async_cleanup
+    zha_gateway.async_enable_debug_mode()
+    await zha_gateway.application_controller.permit(60)
+
+    connection.send_message(websocket_api.result_message(msg['id']))
```

Comment:
There is a new shortcut, `connection.send_result(msg['id'])`

Reply:
I’ll update